### DISCUSSION TITLE
How to debug 'WOULD stop release after commit as the changelog entry is empty for crates'?

### DISCUSSION THREAD

**Discussion by haata** [GitoxideLabs/gitoxide#632] [Q&A] [answered]
I have a couple couple git repos where I'm getting

https://github.com/hid-io/hid-io-core
```
cargo smart-release --update-crates-index hid-io-core
[INFO ] Updating crates-io index
[INFO ] hid-io-core: Tracking top-level crate's changes in multi-crate workspace through 'src/' directory only.
[INFO ] WOULD auto-bump dependent package 'hid-io-protocol' from 0.1.3 to 0.1.4 for publishing
[INFO ] WOULD auto-bump provided package 'hid-io-core' from 0.1.1 to 0.1.2 for publishing
[INFO ] WOULD adjust version constraints in manifest of 1 package as direct dependencies are changing: hid-io-client
[INFO ] WOULD modify existing changelog for 'hid-io-protocol'.
[INFO ] hid-io-core: Tracking top-level crate's changes in multi-crate workspace through 'src/' directory only.
[INFO ] WOULD modify existing changelog for 'hid-io-core'.
[INFO ] Up to 2 changelogs would be previewed if the --execute is set and --no-changelog-preview is unset.
[WARN ] WOULD stop release after commit as the changelog entry is empty for crates: hid-io-protocol, hid-io-core
[WARN ] To fix the changelog manually, run: cargo changelog --write hid-io-core
```

I run the suggested command, commit the changes, but the warning doesn't go away (I've also tried running --allow-dirty instead). The only way I can get smart release to work on this git repo is if I generate the CHANGELOG.md separately (using `--no-changelog`).
I tried looking at the codebase but I'm not quite sure what I'm looking for.
Can anyone give me some pointers on where to investigate?

**Comment by Byron** [comment, marked as answer]
This happens if a changelog entry for a new version would be empty as no conventional commit affected the release. `cargo smart-release` is quite adamant about writing something and I noticed as well that there seems to be no way around that. I still remember that I tried to 'nerf' it once but failed and now think it's probably OK to cater to its will 😅.

All it takes is to execute the suggested command and write a release note for the crates it mentions, `hid-io-protocol`, `hid-io-core` in this case. I usually write `A maintenance release without user-facing changes` right under the `### Unreleased` headline.

I hope that helps.

**Reply by haata** [reply]
Thank you! That was the hint I needed.

For those looking in the future, you just need to make sure your commits follow this format at minimum

```
<something>: <something>
```

Everything else seems optional/freeform.
